### PR TITLE
[ MNIST ] Comparison with Tensorflow

### PR DESCRIPTION
**Pull request by jijoongmoon**
This PR includes:
 . mnist tensorflow example to compare
 . data generation for tensorflow
 . input for mnist ini for small size training set
 . fix no_op_prime for activation layer
 . add sqrtDouble
 . remove conv2d rotate_180

related issues : #133 

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: Applications/mnist/Tensorflow/Training_Keras.py [hunk 1]
```diff
@@ -99,67 +93,73 @@ def datagen( x_data, y_data, batch_size):
...
-    saved_model.summary()
+    tf_grad = optimizer.compute_gradients(tf_loss, var_list = trainable_variables)
     
-    score = saved_model.evaluate(ValVector, ValLabel, verbose=0)
-    print("%s: %.5f%%" % (saved_model.metrics_names[1], score[1]*100))
+    global_step =tf.compat.v1.train.get_or_create_global_step()
+
+    train_op = optimizer.apply_gradients(tf_grad, global_step=global_step)
+
+    update_global_step = tf.assign(global_step, global_step + 1, name = 'update_global_step')
```
Comment (kparichay) [inline Applications/mnist/Tensorflow/Training_Keras.py:122]: this is unused. lets remove it.

File: nntrainer/src/optimizer.cpp [hunk 2]
```diff
@@ -108,6 +113,28 @@ void Optimizer::apply_gradients(std::shared_ptr<UpdatableParam> params,
       break;
     case OptType::adam: {
 
+      // This is implementation of adam from original paper.
+      // This is not deleted intensionally.
```
Comment (kparichay) [inline nntrainer/src/optimizer.cpp:117]: intentionally

**Review by kparichay — APPROVED** [review]
Overall LGTM

File: Applications/mnist/Tensorflow/Training_Keras.py (hunk 1 above)
Comment (jijoongmoon) [reply Applications/mnist/Tensorflow/Training_Keras.py]: fixed.

File: nntrainer/src/optimizer.cpp (hunk 2 above)
Comment (jijoongmoon) [reply nntrainer/src/optimizer.cpp]: fixed.

**Pull request merged**